sdcm/cluster_azure.py

```diff
@@ -84,8 +87,24 @@ def is_spot(self):
     def check_spot_termination(self):
         """Check if a spot instance termination was initiated by the cloud.
         """
-        # todo: fix it
-        return False
+        try:
+            self.wait_ssh_up(verbose=False)
+            result = self.remoter.run(
+                'curl http://169.254.169.254/metadata/scheduledevents?api-version=2020-07-01 -H "Metadata: true"', verbose=False)
+            status = json.loads(result.stdout.strip())
+            if status["DocumentIncarnation"] == self.scheduled_events_document_incarnation:
+                return SPOT_TERMINATION_CHECK_OVERHEAD
+            for event in status["Events"]:
+                self.scheduled_events_document_incarnation = status["DocumentIncarnation"]
+                if event["EventType"] == "Preempt":
+                    message = f"got spot termination event for node: {event['Resources']}. VM will be evicted at: {event['NotBefore']} "
+                    SpotTerminationEvent(node=self, message=message).publish()
+                else:
+                    self.log.warning(f"Unhandled Azure scheduled event: {event}")
+        except Exception as details:  # pylint: disable=broad-except
+            self.log.warning('Error during getting Azure scheduled events: %s', details)
+            return 0
```

Comment:
when having an exception, do we want to still return `0`?
also, the message above could be a at `log.error` level, or not?

Reply:
I did it in the same way as on AWS. Error may happen sometimes e.g due some ssh problem so better to retry ASAP.